### ISSUE TITLE
can't install knex 0.12.4

### ISSUE THREAD

**Issue by ravshansbox** [tgriesser/knex#1736]
ravshan@Ravshans-MacBook-Pro:~/test $ npm install knex  

> knex@0.12.4 postinstall /Users/ravshan/test/node_modules/knex
> node ./scripts/build.js lib "npm run build"

module.js:457
    throw err;
    ^

Error: Cannot find module '/Users/ravshan/test/node_modules/knex/scripts/build.js'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
npm WARN enoent ENOENT: no such file or directory, open '/Users/ravshan/test/package.json'
npm WARN test No description
npm WARN test No repository field.
npm WARN test No README data
npm WARN test No license field.
npm ERR! Darwin 16.0.0
npm ERR! argv "/usr/local/Cellar/node/6.7.0/bin/node" "/usr/local/bin/npm" "install" "knex"
npm ERR! node v6.7.0
npm ERR! npm  v3.10.7
npm ERR! code ELIFECYCLE

npm ERR! knex@0.12.4 postinstall: `node ./scripts/build.js lib "npm run build"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the knex@0.12.4 postinstall script 'node ./scripts/build.js lib "npm run build"'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the knex package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./scripts/build.js lib "npm run build"
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs knex
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls knex
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/ravshan/test/npm-debug.log

**Comment by yutin1987** [comment]
me to, run on `yarn install`

```
liuyutingdeMacBook-Pro-2:server yutin$ yarn install
yarn install v0.15.1
error npm-shrinkwrap.json found. This will not be updated or respected. See [TODO] for more information.
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
[1/2] ⠄ argon2
[2/2] ⠄ knex
[-/2] ⠄ waiting...
[-/2] ⠄ waiting...
error /Users/yutin/Develop/bunninn/client/server/node_modules/knex: Command failed.
Exit code: 1
Command: sh
Arguments: -c node ./scripts/build.js lib "npm run build"
Directory: /Users/yutin/Develop/bunninn/client/server/node_modules/knex
Output:
module.js:457
    throw err;
    ^

Error: Cannot find module '/Users/yutin/Develop/bunninn/client/server/node_modules/knex/scripts/build.js'
    at Function.Module._resolveFilename (module.js:455:15)
    at Function.Module._load (module.js:403:25)
    at Module.runMain (module.js:590:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
    at SpawnError (/Users/yutin/.yarn/lib/errors.js:18:1)
    at ChildProcess.proc.on.code (/Users/yutin/.yarn/lib/util/child.js:107:15)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at maybeClose (internal/child_process.js:877:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:226:5)
info Visit http://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

**Comment by WebReflection** [comment]
same here, this was a **patch** update, it broke things badly.

**Comment by anomaly44** [comment]
came here to report the same issue

**Comment by raymond-h** [comment]
Seems someone put a "postinstall" script in package.json that refers to a script in the "scripts" folder, but that folder is not pushed to the npm registry. Problem lies in that "postinstall" runs after the package is installed, even as a dependency from npm. See 7a04de4b87b611da8de3091cbb6eaeeb2252819b

**Comment by c4d3r** [comment]
for now i used `yarn add knex@0.12.3` which worked

**Comment by WebReflection** [comment]
**breaking news**
`npm install knex@0.12.3` also works since ever!!!

...

**Comment by c4d3r** [comment]
Okay thanks for being so outraged just because you see yarn, next time ill just write npm install knex@0.12.3 and then you are happy, okay? the point was to just give the issuer an alternative to continue his work for now

**Comment by WebReflection** [comment]
@c4d3r I've dropped my comment but even if you had best intentions, I am also sure we all know how to use semver and your comment didn't help much here.

There are two linked mentions that already explain what is wrong and why/when broke.
This patch should merge the suggested fix and publish `0.12.5` to **npm**, flagging as revoked/deprecated/useless the `0.12.4` too.

Now that would help a regular `yarn` install too, isn't it?

**Comment by tarikmlaco** [comment]
Same here :)

**Comment by tgriesser** [comment]
Sorry about that

**Comment by tgriesser** [comment]
Fixed in 0.12.5
